benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/tasks/running/DetermineExecuteExperimentsTask.java

```diff
@@ -55,19 +54,23 @@ public void run() {
       String experimentID = experimentModelDAO.addExperiment(testID);
 
       // get the selection strategy
-      ExperimentSelectionStrategy selectionStrategy =
-          explorationModelDAO.getExperimentSelectionStrategy(testID);
+      SelectionStrategy selectionStrategy = explorationModelDAO.getSelectionStrategy(testID);
 
       // generate the Experiment definition
-      String experimentYaml = selectionStrategy.selectNextExperiment(testID);
-      InputStream experimentYamlInputStream =
-          IOUtils.toInputStream(experimentYaml, StandardCharsets.UTF_8);
+      SelectedExperimentBundle selectedExperimentBundle =
+          selectionStrategy.selectNextExperiment(testID);
+
+      // save exploration point index
+      experimentModelDAO.setExplorationSpaceIndex(experimentID,
+          selectedExperimentBundle.getExplorationSpaceIndex());
 
       // save PE defintion to minio
```

Comment:
Done in https://github.com/benchflow/benchflow/pull/394/commits/3169f96597122c1c799e101f6ba8b10b180ebafd